Find a way to report why a gradle sync fails.

Running GradleSync can fail for several reasons but currently the plugin only prints "Gradle sync task failed" in such cases. 

One way to solve this is to write the output of the sync to a file and provide a command to dump it in the quickfix window, or set an errorformat for it and show the errors.

Ideas are welcome.

This is sample ouput of a sync fail:

```
:ext:vim
vim-project ext
vim-target android-22
:ext:CircularProgressBar:vim
vim-project CircularProgressBar
vim-target android-22
:ext:CircularProgressBar:vim FAILED

FAILURE: Build failed with an exception.

* Where:
Initialization script '/home/ryujin/Projects/vim/vim-android/gradle/init.gradle' line: 15

* What went wrong:
Execution failed for task ':ext:CircularProgressBar:vim'.
> Could not resolve all dependencies for configuration ':ext:CircularProgressBar:androidJacocoAgent'.
   > Could not find org.jacoco:org.jacoco.agent:0.7.4.201502262128.
     Searched in the following locations:
         file:/home/ryujin/Apps/android-sdk/extras/android/m2repository/org/jacoco/org.jacoco.agent/0.7.4.201502262128/org.jacoco.agent-0.7.4.201502262128.pom
         file:/home/ryujin/Apps/android-sdk/extras/android/m2repository/org/jacoco/org.jacoco.agent/0.7.4.201502262128/org.jacoco.agent-0.7.4.201502262128.jar
         file:/home/ryujin/Apps/android-sdk/extras/google/m2repository/org/jacoco/org.jacoco.agent/0.7.4.201502262128/org.jacoco.agent-0.7.4.201502262128.pom
         file:/home/ryujin/Apps/android-sdk/extras/google/m2repository/org/jacoco/org.jacoco.agent/0.7.4.201502262128/org.jacoco.agent-0.7.4.201502262128.jar
     Required by:
         kidint-android.ext:CircularProgressBar:unspecified

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 4.568 secs
```

--- a/autoload/gradle.vim
+++ b/autoload/gradle.vim
@@ -333,7 +333,8 @@ function! gradle#syncCmd()
    \ gradle#findGradleFile(),
    \ "-I",
    \ g:gradle_init_file,
-   \ "vim"
+   \ "vim",
+   \ "2>&1"
    \ ]
 
   return join(l:cmd, ' ')
@@ -415,16 +416,20 @@ function! s:vimTaskHandler(id, data, event) dict
 
   if a:event == 'stdout' || a:event == 'stderr'
     call s:parseVimTaskOutput(self.gradleFile, a:data)
+    call setqflist([], 'a', { 'efm': efm#escapeEfm(efm#buildEfm()), 'lines': a:data, 'title': 'Gradle Sync' })
   elseif a:event == 'exit'
     call s:finishBuilding()
     if a:data != 0
+      call s:showQuickfix()
       call gradle#logi("Gradle sync task failed")
     endif
     call gradle#setup()
   endif
+
 endfunction
 
 function! s:parseVimTaskOutput(gradleFile, result)
+
   for line in a:result
 
     let mlist = matchlist(line, '^gradle-version\s\(\d\+\)\.\(\d\+\).*$')
@@ -449,7 +454,9 @@ function! s:parseVimTaskOutput(gradleFile, result)
     if empty(mlist) == 0 && len(mlist[1]) > 0
       let g:gradle_target_versions[a:gradleFile] = mlist[1]
     endif
+
   endfor
+
 endfunction
 
 ""

